assembl/static2/js/app/components/debate/survey/post.jsx

```diff
@@ -0,0 +1,177 @@
+    .then((sentiments) => {
+      target.setAttribute('class', 'sentiment sentiment-active');
+      this.setState({
+        like: sentiments.data.addSentiment.post.sentimentCounts.like,
+        disagree: sentiments.data.addSentiment.post.sentimentCounts.disagree,
+        mySentiment: sentiments.data.addSentiment.post.mySentiment
+      });
+    }).catch((error) => {
+      this.props.displayAlert('danger', `${error}`);
+    });
+  }
+  deleteSentiment(target) {
+    const { id } = this.props;
+    this.props.deleteSentiment({ variables: { postId: id } })
+    .then((sentiments) => {
+      target.setAttribute('class', 'sentiment');
+      this.setState({
+        like: sentiments.data.deleteSentiment.post.sentimentCounts.like,
+        disagree: sentiments.data.deleteSentiment.post.sentimentCounts.disagree,
+        mySentiment: sentiments.data.deleteSentiment.post.mySentiment
+      });
+    }).catch((error) => {
+      this.props.displayAlert('danger', `${error}`);
+    });
+  }
+  render() {
+    const { postIndex, moreProposals, post, id } = this.props;
+    return (
+      <div className={postIndex < 3 || moreProposals ? 'shown box' : 'hidden box'}>
+        <div className="content">
+          <PostCreator id={id} />
+          <div className="body">{post.body}</div>
+          <div className="sentiments">
+            <div className="sentiment-label">
+              <Translate value="debate.survey.react" />
+            </div>
+            <div
+              className={this.state.mySentiment === 'LIKE' ? 'sentiment sentiment-active' : 'sentiment'}
+              onClick={(event) => { this.handleSentiment(event, 'LIKE'); }}
+            >
+              <Like size={25} />
+            </div>
+            <div
+              className={this.state.mySentiment === 'DISAGREE' ? 'sentiment sentiment-active' : 'sentiment'}
+              onClick={(event) => { this.handleSentiment(event, 'DISAGREE'); }}
+            >
+              <Disagree size={25} />
+            </div>
+          </div>
+        </div>
+        <div className="statistic">
+          <div className="totalSentimentsCount">
+            {this.state.like + this.state.disagree}
+          </div>
+          <div className="sentimentsCountLabel">
+            <Translate value="debate.survey.reactions" />
+          </div>
+          <Circle like={this.state.like} disagree={this.state.disagree} />
+          <div className="stat-sentiment">
+            <div>
+              <div className="min-sentiment">
+                <Like size={15} />&nbsp;<span className="txt">{this.state.like}</span>
+              </div>
+            </div>
+            <div>
+              <div className="min-sentiment">
+                <Disagree size={15} />&nbsp;<span className="txt">{this.state.disagree}</span>
+              </div>
+            </div>
+          </div>
+        </div>
+        <div className="clear">&nbsp;</div>
+      </div>
+    );
+  }
+}
+
+Post.propTypes = {
```

Comment:
It's a good practice to define them for each component. An alternative to prop-types for type checking is flow, which we can consider using.